### PR TITLE
Feature/ennio/model

### PR DESCRIPTION
**Pull request by EnnioEnnio**
Add some description to the empty methods for better orientation.
Add some insights on the Dataset to data_loader.py.
Add a provisional model inspired by LeNet 
(different dimensions and more layers to reduce image size more gradually)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -158,3 +158,10 @@ cython_debug/
 #  and can be added to the global gitignore or merged into this file.  For a more nuclear
 #  option (not recommended) you can uncomment the following to ignore the entire idea folder.
 #.idea/
+
+# SLURM sbatch files
+*.sh
```
Comment (Toalaah) [inline .gitignore:163]: Out of interest, what are these files generated by, specifically? Does it have to do w/ the setup of running the model on a cluster or smth?

File: architecture.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+# Reshaping layer to transform the input from a 2D image to a 4D tensor
+class Reshape(torch.nn.Module):
+    def forward(self, x):
+        return x.view(-1, 3, 256, 256)
+
+
+class LeNetty(nn.Model):
+    """An Interpretation of the LeNet-5 model."""
+
+    def __init__(self, lr=0.1, num_classes=2, momentum=0.9, weight_decay=5e-4, batch_size=128, epochs=200):
```
Comment (Toalaah) [inline architecture.py:20]: Should these parameters not be part of the training loop and not of the model `__init__` itself?

**Review by Toalaah — APPROVED** [review]
lgtm

File: .gitignore (hunk 1 above)
Comment (EnnioEnnio) [reply .gitignore:163]: yeah, you're right. 
The DHC-HPC uses the Simple Linux Utility for Resource Management (SLURM) for scheduling the submitted jobs. You can either run a job live, which can be used for debugging or testing since an active session is required and longer training runs become a problem, or you can define the session parameters and the job to be done in a batch-file (`.sh`) and the scheduler will take care of everything.

File: architecture.py (hunk 2 above)
Comment (EnnioEnnio) [reply architecture.py:20]: I asked myself the same question when I read that it's possible to pass them yesterday. 
I have actually no idea what's best practice here so I tried asking chatgpt which came up with the following explanation: 

> Setting certain parameters, such as learning rate (lr), momentum, weight decay, and batch size, in the __init__ method of your model class can provide convenience and flexibility. Here's why it can be beneficial:
> 
> Modularity: By setting these parameters within the model class, you encapsulate both the architecture and the associated training parameters together. This can make your code more modular and self-contained, allowing you to easily reuse and modify the model without having to pass additional arguments from the training script (train.py).
> Portability: When you save or load a trained model, having the training parameters defined within the model class ensures that all the necessary information is contained within the model itself. This makes it easier to transfer or deploy the model to different environments without having to remember or pass the training parameters separately.

But I'm also not sure and would probably remove them once we're writing the training.
